usecases/scheduler/src/lambda/ssv2_resultProcesser/resultProcesser.py

```diff
@@ -0,0 +1,263 @@
+            diffTotal = currentResults[sheets]['Total'] - previousResults[sheets]['Total']
+        
+        nf = ""
+        ri = ""
+        if len(newFindings):
+            nf = " -- " + ("\n -- ").join(newFindings)
+
+        if len(resolvedItems):
+            ri = " -- " + ("\n -- ").join(resolvedItems)
+
+        data.append([
+            sheets,
+            currentResults[sheets]['High'],
+            currentResults[sheets]['Total'],
+            diffHigh,
+            diffTotal,
+            nf,
+            ri,
+            len(newFindings),
+            len(resolvedItems)
+        ])
+        logging.info(data)
+
+    return data
+
+def formatCompared(compared, hasPreviousObj):
+    row = []
+    news = []
+    resolved = []
+    totalNew = 0
+    totalResolved = 0
+    for trow in compared:
+        tmsg = ""
+        tmsg = "{}: HIGH={} | TOTAL={}".format(trow[0], trow[1], trow[2])
+        if hasPreviousObj:
+            tmsg  = tmsg + " | DIFFHIGH={} | TOTALDIFF={}".format(trow[3], trow[4])
+            if len(trow[5]):
+                totalNew = totalNew + trow[7]
+                news = news + [trow[5]]
+            
+            if len(trow[6]):
+                totalResolved = totalResolved + trow[8]
+                resolved = resolved + [trow[6]]
+
+        row.append(tmsg)
+
+    html = "\n".join(row)
+    html = html + "\n\n"
+    html = html + "\nNew Findings ({}):\n".format(totalNew)
+    html = html + ''.join(news)
+    html = html + "\n\n"
+    html = html + "\nResolved ({}):\n".format(totalResolved)
+    html = html + "\n -- ".join(resolved)
+
+    return html
+
+
+# Not in used, SNS does not support HTML unless pairing it with SES
+def formatComparedHTML(compared, hasPreviousObj):
+
+    template = "<table><thead><tr>{}</tr></thead><tbody>{}</tbody></table>"
+
+    thead = ["Services", "# High", "# Total"]
+    if hasPreviousObj:
+        thead.append('# High (diff)')
+        thead.append('# Total (diff)')
+        thead.append("New Findings")
+        thead.append("Resolved")
+
+    theadHTML = ''.join(["<th>{}</th>".format(x) for x in thead])
+    
+    tbodyHTML = []
+    for row in compared:
+        trow = [row[0], row[1], row[2]]
+        if hasPreviousObj:
+            trow.append(row[3])
+            trow.append(row[4])
+            trow.append(row[5])
+            trow.append(row[6])
+        
+        tbodyStr = ''.join(["<td>{}</td>".format(x) for x in trow])
+        tbodyHTML.append("<tr>{}</tr>".format(tbodyStr))
+    
+    tbodyHTML = ''.join(tbodyHTML)
+    return template.format(theadHTML, tbodyHTML)
+
+def checkIfPreviousScanFound():
+    for acct, info in accounts.items():
+        if not info.get('previousRun'):
+            return False
+
+    return True
+    pass
+
+def sendSnsEmail(sns, configId, html):
+    topic = snsPrefix + '-' + configId
+    rrr = sns.list_topics()
+    topicArn = [tp['TopicArn'] for tp in sns.list_topics()['Topics'] if topic in tp['TopicArn']]
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `list_topics` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)